libcircle.spec

```diff
@@ -0,0 +1,191 @@
+Name:    libcircle
+%global maj_ver 0.3
+Version: %{maj_ver}.0
+Release: 1%{?dist}
+
+Source: https://github.com/hpc/%{name}/releases/download/v%{maj_ver}/%{name}-%{version}.tar.gz
+URL: http://hpc.github.io/libcircle/
+Summary: A library used to distribute workloads
+License: BSD
+
+%global openmpi_ver openmpi3
+
+%if 0%{?fedora}
+%global openmpi_ver openmpi
+%endif
+
+BuildRequires:  check-devel
+BuildRequires:  doxygen
+BuildRequires:  graphviz
+
+%description
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+%package %{openmpi_ver}
+Summary:        Libcircle Open MPI libraries
+BuildRequires:  %{openmpi_ver}-devel
+
+%description %{openmpi_ver}
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+libcircle compiled with Open MPI
+
+%package mpich
+Summary:        Libcircle MPICH libraries
+BuildRequires:  mpich-devel environment-modules
+
+%description mpich
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+libcircle compiled with MPICH
+
+%package doc
+Summary:        Documentation for libcircle
+BuildArch:      noarch
+
+%description doc
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains documentation for libcircle
+
+%package %{openmpi_ver}-devel
+Summary:    Development headers and libraries for Open MPI libcircle
+Requires:   %{name}-%{openmpi_ver}%{?_isa} = %{version}-%{release}
+
+%description %{openmpi_ver}-devel
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains development headers and libraries for Open
+MPI ibcircle
+
+%package mpich-devel
+Summary:    Development headers and libraries for MPICH libcircle
+Requires:   %{name}-mpich%{?_isa} = %{version}-%{release}
+
+%description mpich-devel
+A simple interface for processing workloads using an automatically
+distributed global queue.
+
+This package contains development headers and libraries for
+MPICH libcircle
+
+%prep
+%setup -q
+
+%build
+mkdir %{openmpi_ver} mpich
+%global _configure ../configure
+
+pushd %{openmpi_ver}
+%if "%{openmpi_ver}" == "openmpi"
+%{_openmpi_load}
+%else
+%{_openmpi3_load}
+%endif
+%configure --enable-doxygen --enable-tests --disable-static --libdir="${MPI_LIB}" --includedir="${MPI_INCLUDE}"
+%make_build
+%if "%{openmpi_ver}" == "openmpi"
+%{_openmpi_unload}
+%else
+%{_openmpi3_unload}
+%endif
+popd
+
+pushd mpich
+%{_mpich_load}
+%configure --enable-tests --disable-static --libdir="${MPI_LIB}" --includedir="${MPI_INCLUDE}"
+%make_build
+%{_mpich_unload}
+popd
```

Comment:
The above two builds can be DRYed out into a loop such as is done in https://github.com/daos-stack/hdf5-vol-daos/blob/759b51c503a0fff053eb84a16925bfd2d66060e3/hdf5-vol-daos.spec#L132-L148.  That same file also demonstrates how to define the load/unload macros to make the `%build` and `%install` tidier.

I have actually been considering that we want to create our own mpi-builder RPM that installs those macros so that all of these multi-mpi building RPMs don't have to keep repeating those macros.

But in any case, we should at least be consistent with how we do this multi-mpi building across all of our packages and the hdf5-vol-daos one is a good model.